exchanges/dydx/dydx_wrapper.go

```diff
@@ -0,0 +1,927 @@
+		dy.PrintEnabledPairs()
+	}
+
+	if !dy.GetEnabledFeatures().AutoPairUpdates {
+		return
+	}
+
+	err := dy.UpdateTradablePairs(context.TODO(), false)
+	if err != nil {
+		log.Errorf(log.ExchangeSys,
+			"%s failed to update tradable pairs. Err: %s",
+			dy.Name,
+			err)
+	}
+}
+
+// FetchTradablePairs returns a list of the exchanges tradable pairs
+func (dy *DYDX) FetchTradablePairs(ctx context.Context, a asset.Item) (currency.Pairs, error) {
+	if !dy.SupportsAsset(a) {
+		return nil, fmt.Errorf("asset type of %s is not supported by %s", a, dy.Name)
+	}
+	instruments, err := dy.GetMarkets(ctx, "")
+	if err != nil {
+		return nil, err
+	}
+	pairs := make(currency.Pairs, len(instruments.Markets))
+	count := 0
+	for key := range instruments.Markets {
+		cp, err := currency.NewPairFromString(key)
+		if err != nil {
+			return nil, err
+		}
+		pairs[count] = cp
+		count++
+	}
+	return pairs, nil
+}
+
+// UpdateTradablePairs updates the exchanges available pairs and stores
+// them in the exchanges config
+func (dy *DYDX) UpdateTradablePairs(ctx context.Context, forceUpdate bool) error {
+	pairs, err := dy.FetchTradablePairs(ctx, asset.Spot)
+	if err != nil {
+		return err
+	}
+	return dy.UpdatePairs(pairs, asset.Spot, false, forceUpdate)
+}
+
+// UpdateTicker updates and returns the ticker for a currency pair
+func (dy *DYDX) UpdateTicker(ctx context.Context, p currency.Pair, assetType asset.Item) (*ticker.Price, error) {
+	fPair, err := dy.FormatExchangeCurrency(p, assetType)
+	if err != nil {
+		return nil, err
+	}
+	stats, err := dy.GetMarketStats(ctx, fPair.String(), 1)
+	if err != nil {
+		return nil, err
+	}
+	if len(stats) == 0 {
+		return nil, fmt.Errorf("missing ticker data for instrument %s", fPair.String())
+	}
+	for key, tick := range stats {
+		if !fPair.IsEmpty() && !strings.EqualFold(fPair.String(), key) {
+			continue
+		}
+		cp, err := currency.NewPairFromString(tick.Market)
+		if err != nil {
+			return nil, err
+		}
+		err = ticker.ProcessTicker(&ticker.Price{
+			Pair:         cp,
+			High:         tick.High,
+			Low:          tick.Low,
+			Close:        tick.Close,
+			Open:         tick.Open,
+			Volume:       tick.BaseVolume,
+			QuoteVolume:  tick.QuoteVolume,
+			ExchangeName: dy.Name,
+			AssetType:    assetType,
+		})
+		if err != nil {
+			return nil, err
+		}
+		if !fPair.IsEmpty() && cp.Equal(fPair) {
+			return ticker.GetTicker(dy.Name, p, assetType)
+		}
+	}
+	return ticker.GetTicker(dy.Name, p, assetType)
+}
+
+// UpdateTickers updates all currency pairs of a given asset type
+func (dy *DYDX) UpdateTickers(ctx context.Context, assetType asset.Item) error {
+	pairs, err := dy.GetEnabledPairs(assetType)
+	if err != nil {
+		return err
+	}
+	if !dy.SupportsAsset(assetType) {
+		return fmt.Errorf("%w %v", asset.ErrNotSupported, assetType)
+	}
+	stats, err := dy.GetMarketStats(ctx, "", 30)
+	if err != nil {
+		return err
+	}
+
+	for x := range stats {
+		pair, err := currency.NewPairFromString(stats[x].Market)
```

Comment:
Use `pairs.DeriveFrom` and filter delimiters if they are contained. It's slightly more efficient.